scripts/install-cni.sh

```diff
@@ -405,40 +419,30 @@ cilium_health_check() {
     http://localhost:"${healthz_port}"/healthz
 }
 
-# Try to decouple RUN_CNI_WATCHDOG and ENABLE_CILIUM_PLUGIN; don't assume
-# ENABLE_CILIUM_PLUGIN is set whenever RUN_CNI_WATCHDOG is set.
-if [[ "${RUN_CNI_WATCHDOG:-}" != "true" ]]; then
-
-  # In non-watchdog mode, we must exit after writing CNI config.
-  echo "Not running CNI watchdog. Will exit as soon as CNI config is written."
-
-  if [[ "${ENABLE_CILIUM_PLUGIN:-}" == "true" ]]; then
-    if cilium_health_check "${CILIUM_HEALTH_MAX_WAIT_TIME:-600}"; then
-      echo "Cilium healthz reported success."
-    else
-      echo "Cilium not yet ready. Continuing anyway."
-    fi
+# Cilium health check logic before watchdog and CNI STATUS API are introduced.
+cilium_wait_or_ignore() {
+  if cilium_health_check "${CILIUM_HEALTH_MAX_WAIT_TIME:-600}"; then
+    echo "Cilium healthz reported success."
+  else
+    echo "Cilium not yet ready. Continuing anyway."
   fi
+}
 
+write_and_success() {
```

Comment:
Maybe changing this to write_cni_conf and making a separate call to success is cleaner.

Reply:
It's already merged. Let's leave it as-is.